python-stdlib/unittest-discover/unittest/__main__.py

```diff
@@ -121,22 +121,27 @@ def discover_main():
     ):
         # No args, or `python -m unittest discover ...`.
         result = _discover(runner)
-        failures += result.failuresNum or result.errorsNum
     else:
+        result = TestResult()
         for test_spec in sys.argv[1:]:
             try:
                 os.stat(test_spec)
                 # File exists, strip extension and import with its parent directory in sys.path.
                 dirname, module_name = _dirname_filename_no_ext(test_spec)
-                result = _run_test_module(runner, module_name, dirname)
+                res = _run_test_module(runner, module_name, dirname)
             except OSError:
                 # Not a file, treat as named module to import.
-                result = _run_test_module(runner, test_spec)
+                res = _run_test_module(runner, test_spec)
 
-            failures += result.failuresNum or result.errorsNum
+            result += res
+
+    if not result.testsRun:
+        # If tests are run their results are already printed.
+        # Ensure an output is still given if no tests are found.
+        result.printResults()
```

Comment:
Instead of exposing `printResults()` as a separate function (which isn't in CPython...), would it be possible (and simpler) to just call `runner.run(TestSuite())` here?

Reply:
Nice idea, thanks!
I've made this change and yes it does appear to work correctly.